integration/e2e_test.go

```diff
@@ -866,6 +866,126 @@ func TestE2E_ConcurrentTransactionSubmission(t *testing.T) {
 	}
 }
 
+func TestE2E_API_GetBlocksWithFullTransactions(t *testing.T) {
```

Comment:
I would say an e2e test to just test this feature of the one API is an overkill. The reason why I feel this way is that these e2e tests might get really bloated with time and we should also think about maintaining tests. My suggestion is to incorporate this test in one of the above existing e2e test flows and just do basic comparison (comparing to hardcoded values of hashes and hex encoded values is also a bit problematic because they might change). So maybe check that transactions are included by checking the length of transactions and block number is indeed correct etc. Values we can check with existing variables. For more detailed tests I would say we should start write unit tests of the API itself not part of e2e tests.

Reply:
Sounds fair :+1: Adding an e2e test also takes up some seconds, to start a new emulator and all the necessary components. I will update to what you suggested, and in a follow-up PR I will try to include client testing in CI (with web3.js).

Reply:
@sideninja I have removed the dedicated test case for the new functionality, and put a small version of these assertions in the existing test case.